int16[4] data
